org.jacoco.core/src/org/jacoco/core/internal/analysis/filter/IFilterOutput.java

```diff
@@ -50,10 +51,50 @@ public interface IFilterOutput {
 	 *
 	 * @param source
 	 *            instruction which branches should be replaced
-	 * @param newTargets
-	 *            new targets of branches
+	 * @param newBranches
+	 *            new branches
+	 * @see Replacements
 	 */
 	void replaceBranches(AbstractInsnNode source,
-			Set<AbstractInsnNode> newTargets);
+			Iterable<Collection<InstructionBranch>> newBranches);
```

Comment:
The parameter type `Iterable<Collection<InstructionBranch>>` looks strange without further explanation. A wonder whether you could use `Replacements` as a type here. In nearly all use case the method is called with:

    replaceBranches(..., replacements.values());
    
Then it would be 

    replaceBranches(..., replacements);